spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,343 @@
+          expect(assigns(:associated_equipment_models)).to eq([mod1])
         end
 
-        it_behaves_like 'calculates availability correctly', 2
+        it 'limits @associated_equipment_models to maximum 6' do
+          model.associated_equipment_models =
+            FactoryGirl.create_list(:equipment_model, 7)
+          get :show, id: model
+          expect(assigns(:associated_equipment_models).size).to eq(6)
+        end
       end
+      USER_ROLES.each { |type| it_behaves_like 'GET show success', type }
 
-      context 'with upcoming reservation' do
+      context 'with admin user' do
         before do
-          FactoryGirl.create(:valid_reservation,
-                             equipment_model: @model,
-                             start_date: Time.zone.today + 1.day,
-                             due_date: Time.zone.today + 1.day)
+          FactoryGirl.create_pair(:equipment_item, equipment_model: model)
+          mock_user_sign_in(mock_user(:admin))
+        end
+        let!(:missed) do
+          FactoryGirl.create(:missed_reservation, equipment_model: model)
+        end
+        let!(:starts_today) do
+          FactoryGirl.create(:reservation, equipment_model: model,
+                                           start_date: Time.zone.today,
+                                           due_date: Time.zone.today + 2.days)
+        end
+        let!(:starts_this_week) do
+          FactoryGirl.create(:reservation, equipment_model: model,
+                                           start_date: Time.zone.today + 2.days,
+                                           due_date: Time.zone.today + 4.days)
+        end
+        let!(:starts_next_week) do
+          FactoryGirl.create(:reservation, equipment_model: model,
+                                           start_date: Time.zone.today + 9.days,
+                                           due_date: Time.zone.today + 11.days)
+        end
+        it 'includes @pending reservations' do
+          get :show, id: model
+          expect(assigns(:pending)).to \
+            match_array([starts_today, starts_this_week])
         end
-
-        it_behaves_like 'calculates availability correctly', 1
       end
 
-      context 'with overdue reservation' do
-        before do
-          FactoryGirl.create(:overdue_reservation,
-                             equipment_model: @model,
-                             start_date: Time.zone.today - 1.day,
-                             due_date: Time.zone.today - 1.day,
-                             equipment_item: @model.equipment_items.first)
-          # needed to persist overdue_count
-          @model.save
+      context 'availability calendar' do
+        before(:each) do
+          FactoryGirl.create_pair(:equipment_item, equipment_model: model)
+          mock_user_sign_in
         end
 
-        it_behaves_like 'calculates availability correctly', 1
-      end
+        shared_examples_for 'calculates availability correctly' do |expected|
```

Comment:
while testing availability calculations are important, this kind of tests really belong in the model spec. We should make sure that the model spec adequately tests the availability methods and remove these.